rust/cloud-storage/properties/src/domain/service.rs

```diff
@@ -13,4 +13,12 @@ pub trait PropertiesService: Send + Sync + 'static {
         entity_id: &str,
         entity_type: EntityType,
     ) -> impl Future<Output = Result<(), Self::Err>> + Send;
+
+    /// Check if an entity's status system property is "Completed".
+    /// Returns `false` if the entity doesn't have a status property attached.
+    fn is_system_property_status_complete(
```

Comment:
I don't really like the idea of having this pattern as it extrapolates to each system property requiring a method like this.

I think it would be better if we had a service method that simply takes a status property uuid (SystemPropertyKey enum) and the entity id/type and returned out some Option<T> where T is the expected type of the property.

Also this should return `Option<T>` rather than a bool since it may not exist for the property